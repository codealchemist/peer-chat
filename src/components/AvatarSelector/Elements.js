import styled from 'styled-components'
import Icon from '@material-ui/core/Icon'

export const styles = theme => ({
  paper: {
    position: 'absolute',
    width: theme.spacing.unit * 50,
    maxWidth: '95%',
    color: 'black',
    boxShadow: theme.shadows[5],
    outline: 'none',
    top: '20%',
    left: 0,
    right: 0,
    margin: 'auto',
    textAlign: 'center',
    height: 'calc(80% - 10px)',
    overflow: 'scroll'
  }
})

export const Wrapper = styled.div`
  width: 56px;
  height: 56px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
`

export const ModalContent = styled.div`
  background: rgba(255, 255, 255, 0.8);
  padding: 10px;
`

export const Title = styled.h4`
  margin-top: 0;
`

export const IconsList = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
`

export const StyledIcon = styled(Icon)`
  cursor: pointer;
  :hover {
    color: white;
  }
`
