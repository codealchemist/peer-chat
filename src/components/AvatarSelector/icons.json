[
  "accessibility_new",
  "account_box",
  "account_circle",
  "android",
  "aspect_ratio",
  "autorenew",
  "backup",
  "bookmark",
  "bug_report",
  "build",
  "camera_enhance",
  "card_giftcard",
  "change_history",
  "check_circle",
  "code",
  "copyright",
  "credit_card",
  "dashboard",
  "description",
  "done",
  "drag_indicator",
  "eject",
  "euro_symbol",
  "explore",
  "extension",
  "face",
  "favorite",
  "fingerprint",
  "grade",
  "help",
  "home",
  "hourglass_empty",
  "https",
  "language",
  "lock",
  "lock_open",
  "motorcycle",
  "offline_bolt",
  "pan_tool",
  "pets",
  "power_settings_new",
  "pregnant_woman",
  "room",
  "rowing",
  "stars",
  "verified_user",
  "explicit",
  "flash_on"
]
