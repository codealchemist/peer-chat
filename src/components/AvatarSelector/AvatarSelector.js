import React from 'react'
import { connect } from 'react-redux'
import { compose } from 'redux'
import { setAvatar } from 'store/user/actions'
import { withStyles } from '@material-ui/core/styles'
import Modal from '@material-ui/core/Modal'
import Icon from '@material-ui/core/Icon'
import { Wrapper, IconsList, StyledIcon, styles } from './Elements'
import icons from './icons.json'

export class AvatarSelector extends React.PureComponent {
  state = {
    isModalOpen: false
  }

  openModal = () => {
    this.setState({ isModalOpen: true })
  }

  closeModal = () => {
    this.setState({ isModalOpen: false })
  }

  selectAvatar = avatar => {
    console.log('selected avatar', avatar)
    this.props.setAvatar({ avatar })
    this.closeModal()
  }

  render() {
    const { classes, avatar } = this.props
    return (
      <>
        <Wrapper id="open-modal-button" onClick={this.openModal}>
          <Icon style={{ fontSize: '50px' }}>{avatar}</Icon>
        </Wrapper>

        <Modal open={this.state.isModalOpen} onClose={this.closeModal}>
          <div className={classes.paper}>
            <h4>SELECT AVATAR</h4>
            <IconsList>
              {icons.map(icon => (
                <StyledIcon
                  key={icon}
                  style={{ fontSize: '50px' }}
                  onClick={() => this.selectAvatar(icon)}
                >
                  {icon}
                </StyledIcon>
              ))}
            </IconsList>
          </div>
        </Modal>
      </>
    )
  }
}

const mapStateToProps = state => ({
  avatar: state.user.avatar
})

const mapActionsToProps = {
  setAvatar
}

export default compose(
  withStyles(styles),
  connect(
    mapStateToProps,
    mapActionsToProps
  )
)(AvatarSelector)
