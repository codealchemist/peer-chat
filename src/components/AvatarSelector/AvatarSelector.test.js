import React from 'react'
import { mount, shallow } from 'enzyme'
import AvatarSelector from './index'
import { AvatarSelector as Component } from './AvatarSelector'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

const mockStore = configureStore()
const store = mockStore({
  user: {
    avatar: 'rock-it'
  }
})

describe('AvatarSelector', () => {
  it('should render connected component', () => {
    const wrapper = mount(
      <Provider store={store}>
        <AvatarSelector />
      </Provider>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should open modal', () => {
    const classes = {
      paper: {}
    }
    const wrapper = shallow(<Component classes={classes} />)

    const instance = wrapper.instance()
    instance.setState = jest.fn()

    const iconButton = wrapper.find('#open-modal-button')
    iconButton.simulate('click')
    expect(instance.setState).toHaveBeenCalledWith({ isModalOpen: true })
  })

  it('should close modal', () => {
    const classes = {
      paper: {}
    }
    const wrapper = shallow(<Component classes={classes} />)

    const instance = wrapper.instance()
    instance.setState = jest.fn()
    instance.closeModal()
    expect(instance.setState).toHaveBeenCalledWith({ isModalOpen: false })
  })

  it('should select avatar and close modal', () => {
    const classes = {
      paper: {}
    }
    const wrapper = shallow(
      <Component classes={classes} setAvatar={jest.fn()} />
    )

    const instance = wrapper.instance()
    instance.setState = jest.fn()
    instance.selectAvatar('jazz')
    expect(instance.props.setAvatar).toHaveBeenCalledWith({ avatar: 'jazz' })
    expect(instance.setState).toHaveBeenCalledWith({ isModalOpen: false })
  })
})
