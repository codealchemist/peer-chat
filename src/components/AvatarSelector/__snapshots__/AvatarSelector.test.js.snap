// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarSelector should render connected component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <WithStyles(Connect(AvatarSelector))>
    <Connect(AvatarSelector)
      classes={
        Object {
          "paper": "Connect-AvatarSelector--paper-1",
        }
      }
    >
      <AvatarSelector
        avatar="rock-it"
        classes={
          Object {
            "paper": "Connect-AvatarSelector--paper-1",
          }
        }
        setAvatar={[Function]}
      >
        <styled.div
          id="open-modal-button"
          onClick={[Function]}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": true,
                  "lastClassName": "etOIPa",
                  "rules": Array [
                    "
  width: 56px;
  height: 56px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            id="open-modal-button"
            onClick={[Function]}
          >
            <div
              className="sc-bdVaJa etOIPa"
              id="open-modal-button"
              onClick={[Function]}
            >
              <WithStyles(Icon)
                style={
                  Object {
                    "fontSize": "50px",
                  }
                }
              >
                <Icon
                  classes={
                    Object {
                      "colorAction": "MuiIcon-colorAction-5",
                      "colorDisabled": "MuiIcon-colorDisabled-7",
                      "colorError": "MuiIcon-colorError-6",
                      "colorPrimary": "MuiIcon-colorPrimary-3",
                      "colorSecondary": "MuiIcon-colorSecondary-4",
                      "fontSizeInherit": "MuiIcon-fontSizeInherit-8",
                      "fontSizeLarge": "MuiIcon-fontSizeLarge-10",
                      "fontSizeSmall": "MuiIcon-fontSizeSmall-9",
                      "root": "MuiIcon-root-2",
                    }
                  }
                  color="inherit"
                  component="span"
                  fontSize="default"
                  style={
                    Object {
                      "fontSize": "50px",
                    }
                  }
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-2"
                    style={
                      Object {
                        "fontSize": "50px",
                      }
                    }
                  >
                    rock-it
                  </span>
                </Icon>
              </WithStyles(Icon)>
            </div>
          </StyledComponent>
        </styled.div>
        <WithStyles(Modal)
          onClose={[Function]}
          open={false}
        >
          <Modal
            BackdropComponent={[Function]}
            classes={
              Object {
                "hidden": "MuiModal-hidden-12",
                "root": "MuiModal-root-11",
              }
            }
            closeAfterTransition={false}
            disableAutoFocus={false}
            disableBackdropClick={false}
            disableEnforceFocus={false}
            disableEscapeKeyDown={false}
            disablePortal={false}
            disableRestoreFocus={false}
            hideBackdrop={false}
            keepMounted={false}
            manager={
              ModalManager {
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "modals": Array [],
              }
            }
            onClose={[Function]}
            open={false}
          />
        </WithStyles(Modal)>
      </AvatarSelector>
    </Connect(AvatarSelector)>
  </WithStyles(Connect(AvatarSelector))>
</Provider>
`;
